lib/OpenQA/Git.pm

```diff
@@ -170,4 +174,38 @@ sub is_workdir_clean ($self) {
     return $r->{status};
 }
 
+sub cache_ref ($self, $ref, $url, $relative_path, $output_file, $allow_arbitrary_url_fetch = 1) {
+    OpenQA::App->singleton->log->debug("Checking out $relative_path from url '" . ($url // '') . "' rev '" . ($ref // '') . "' to $output_file");
+    # checkout git versioned file <$relative_path> of from repo at version <$ref> and write it to <$output_file>
+    # returns undef on success - else the error message
+    if (-f $output_file) {
+        eval { path($output_file)->touch };
+        return $@ ? $@ : undef;
+    }
+
+    # Now check if the ref is already present in the repo
+    my @git = $self->_prepare_git_command(1);
+    my $gcf_res = run_cmd_with_log_return_error [@git, 'cat-file', '-t', $ref];
```

Comment:
why not this?
```suggestion
    my $gcf_res = $self->_run_cmd(['cat-file', '-t', $ref]);
```

Reply:
Actually I wrote the `check_sha` method for checking if a ref exists. Maybe that can be reused instead.

Reply:
That one uses `git rev-parse` internally, which does exit with 0 despite providing a non existing object.
So we can't use it here.

Reply:
Well, `check_sha` returns 1 if the ref exists, and 0 if it doesn't. Isn't this what we want?

Reply:
As you can see here, `git rev-parse` does not work as it would be required here. So I guess `check_sha` also doesn't work as intended.

```
[dheidler@midas opi (master)]$ git cat-file -t dc8b5b980544ee9924845c1a7b281b93dda48738
commit
[dheidler@midas opi (master)]$ git rev-parse --verify -q dc8b5b980544ee9924845c1a7b281b93dda48738 ; echo $?
dc8b5b980544ee9924845c1a7b281b93dda48738
0
[dheidler@midas opi (master)]$ git cat-file -t dc8b5b980544ee9924845c1a7b281b93dda48123
Schwerwiegend: git cat-file: could not get object info
[dheidler@midas opi (master)]$ git rev-parse --verify -q dc8b5b980544ee9924845c1a7b281b93dda48123 ; echo $?
dc8b5b980544ee9924845c1a7b281b93dda48123
0

Reply:
And here an example with calling `check_sha` using an invalid hash and it still returning 1:
```perl
PERL5LIB=. ips
This is Interactive Perl shell
perl> use OpenQA::Git;
perl>
perl> $g = OpenQA::Git->new(app => undef, dir => "/home/dheidler/devel/opi");
OpenQA::Git=HASH(0x555c59df42a8)
perl> $g->check_sha("dc8b5b980544ee9924845c1a7b281b93dda48123");
[INFO] Running cmd: git -C /home/dheidler/devel/opi rev-parse --verify -q dc8b5b980544ee9924845c1a7b281b93dda48123
[DEBUG] dc8b5b980544ee9924845c1a7b281b93dda48123

[DEBUG]
[INFO] cmd returned 0
1

Reply:
Oh, I wasn't aware of that. That's a really weird feature then! Apparently it simply succeeds if the ref has 40 characters.
You should do more reviews; then you could have told me already when I added `check_sha` in the first place :)

In this case I suggest we create a new function and use it in both places. But I can do that afterwards.

There is still one change I'd like to suggest though, see https://github.com/os-autoinst/openQA/pull/6190/files#r1973855456